expression/util.go

```diff
@@ -599,13 +606,19 @@ func SubstituteCorCol2Constant(expr Expression) (Expression, error) {
 			}
 			return &Constant{Value: val, RetType: x.GetType()}, nil
 		}
-		var newSf Expression
+		var (
+			err   error
+			newSf Expression
+		)
 		if x.FuncName.L == ast.Cast {
 			newSf = BuildCastFunction(x.GetCtx(), newArgs[0], x.RetType)
+		} else if x.FuncName.L == ast.Grouping {
+			newSf = x.Clone()
+			newSf.(*ScalarFunction).GetArgs()[0] = newArgs[0]
 		} else {
-			newSf = NewFunctionInternal(x.GetCtx(), x.FuncName.L, x.GetType(), newArgs...)
```

Comment:
There raise a question that, do we need to throw panic for the `NewFunctionInternal` since it doesn't init meta for `grouping`?

Reply:
1: I just mark it as deprecated here, the later usage will be notified to change. (blocking for coming)
2: for old code logic that encountering grouping function without initialization here. (prevent for existence)
* the terror.Log(err) in `newFunctionInternal` will log out the error trace in planner layer.
* client side will got evaluation error for grouping function in execution layer.

If necessary, we can definitely  throw the error out OR make a strong assertion there:
1: migrate all `newFunctionInternal` ref to `newFunction`
2: change Terror.Log(err) as Assert(err) to be not nil to panic